### DISCUSSION TITLE
회고 작성해서 올려주시면 감사합니다.  KPT 회고

### DISCUSSION THREAD

**Discussion by Sam1000won** [DJMLteam2/DP_main#37] [Announcements]
K : 잘 해와서 유지하고 싶은 것
P : 어려움을 느껴서 개선하고 싶은 것
T : 구체적인 시도할 내용

**Comment by 8trider** [comment]
K : 난관이 많았는데 탈주 없이 끝까지 해난 끈기력과 멘탈 회복 탄성력.

P : 자바와 자바스크립트에 대한 부족한 이해도.

T : 코드 리팩토링 - 유지 보수를 쉽게, 그래고 복습하는 의미에서 코드를 보다 깔끔하게 정리할 계획입니다.
자바와 스크립트를 더 공부하면서 view단의 mustache를 업그레이드 해보고 싶습니다.

**Reply by Sam1000won** [reply]
유감...

**Comment by Sam1000won** [comment]
K : 하다가 잘 안될 때 팀원들과 소통해서 안되는 부분을 지원 받거나 팀의 분위기를 띄우는 것을 계속 유지하고 싶음
P : 자바스크립트와 자바의 대한 이해도 부족과 백엔드 아키택처를 설계하고 작성하는 부분에서 많이 부족하다는 것을 느꼇고 Js 단순한 코드 오타나 오류로 인해서 계속 오류나는 부분이 있어서 헤매이는 것을 느낌 그래서 코드를 꼼꼼히 보고 복하면서 오타가 적게 하고 싶고 자바 스프링과 자바를 좀 더 공부해서 언어의 이해도를 높이고 싶음 또한 아키택처를 프로젝트를 하면서 처음 설계하고 해보았는데 어떤식으로 구성을하고 효율적으로 구성하고 설계하는 능력을 개선하고 키우고 싶음.
T : 자바, 자바스프링을 사용하는 언어포맵을 더 사용하면서 구조나 이해도를 높이고 아키택쳐를 구성하는 원리나 어떤 부분에서 작동을 해야하는지 어디 부분이 필요하고 수정을 반복하면서 프로그램의 큰그림을 그릴 수 있도록 노력해야 하는 부분을 익히고 노력할 수 있도록 해야겠다.

**Comment by tjkpolisher** [comment]
# Keep
* 큰 난관이 여러 번 있었음에도 불구하고 꾸준하게 노력하며 끝내 프로젝트를 완성시킨 것.
* 높은 출석률. 이거 진짜 큰 거 같음. 항상 두세 명이라도 나와 있으니까 파이널 프로젝트의 맥이 심각하게 끊어지지는 않음.
* 그래도 조원 각자가 익숙한 분야가 하나라도 있었기 때문에 서로 도움을 주고받는 것이 상대적으로 용이했음.
# Problem
* 백엔드와 프론트엔드 도메인 자체에 대한 숙련도. 특히 프론트엔드 또한 만만한 분야가 아님을 이번 프로젝트 중 뼈저리게 느낌.
* 오픈 API 이용 시 키 관리 등 보안 문제. 커밋/푸시할 때마다 수동으로 관리하려니까 많이 귀찮았음.
* 프로젝트 초기 EDA의 부재. 모델 선정이라는 주제에 너무 매몰되어 소홀히 했던 감이 있는데, 사실은 EDA가 더 중요했었다.
# Try
* 꼭 풀스택 개발자가 아니더라도 IT 개발의 모든 분야(프론트엔드, 백엔드, 보안, 데이터 분석, 데이터 엔지니어링, 프로젝트 배포 등등)를 얕고 넓게 배워 볼 필요는 있다고 생각함. 그래야 큰 틀에서의 아키텍처 수립에 애를 먹지 않고 방향성을 잘 잡을 수 있을 것 같음.
>사실은 그 부분이 이번 데이터 엔지니어/MLOps 과정의 진정한 개설 취지가 아니었나 싶음(그걸 감안하면 6개월은 솔직히 짧은 듯).  

* 풀 리퀘스트를 포함한 깃허브의 다른 협업 기능, 더 나아가서 노션과 슬랙 등의 협업 툴을 더 능숙하고 유용하게 다룰 수 있는 방법을 더 연구하면 좋을 듯. 시간적 여유가 있었다면 코드 뿐만 아니라 이런 업무 방법론을 정리해 둔 리포지터리도 같이 참고했으면 좋았을 것 같음.
* 개인적인 바램: 시간이 모자라서 시도할 수 없었던 다른 기법들도 각자 사이드 프로젝트에서 한 번씩 적용해보면 실력 향상에 도움이 될 것 같습니다. NLP 모델과 트랜스포머 모델의 파인 튜닝, 구글 지도 API 내의 다른 API를 이용한 직선 경로 또는 실제 도로 경로 출력 등, 그 외 각자 아쉬웠던 부분을 정리해서 여유 있게 구현해보고 싶습니다.

**Comment by doxgxxn** [comment]
K : 거의 모든 서비스를 다 만져볼 수 있어서 좋았음 
P : 협업이 효율적으로 이루어진 것 같지 않음 한가지에 여러명이 몰린다고 잘되는 것도 아니고 각자 따로해도 그렇게 낫지는 않음
T : 궁금하거나 시도해보고 싶었던 곳들이 많은데 시간 여유상 시도하지 못했던 것이 많음
